src/main/io/vtx_smartaudio.c

```diff
@@ -804,16 +911,29 @@ bool vtxSAGetPitMode(uint8_t *pOnOff)
     return true;
 }
 
+bool vtxSAGetFreq(uint16_t *pFreq)
+{
+    if (!vtxSAIsReady())
```

Comment:
I really dislike all this protection code. It implies the initialisation sequence is either not under control or not fully understood. I realise some of this is inherited from previous code - I hadn't spotted that it was introduced in earlier PRs.

As a general principle device functions should not be called until the device is initialised. As long as that rule is followed these "IsReady" functions are not necessary. Having "IsReady" functions causes code bloat, but more importantly is a source of future errors, since the fact that they exist means it is possible that device functions may be called before the device is ready.

It really is a fundamental principle that the initialisation must be fully understood and fully deterministic. Again, I realise that you are building on inherited errors and so may have limited choice, but we really need to sort this out.

Reply:
The "protection code" is coming from the existing versions, it's not something new introduced in this PR, so sorting it out should be done in a different PR or issue.